lxd/ovn.go

```diff
@@ -0,0 +1,115 @@
+package main
+
+import (
+	"context"
+	"net"
+	"strings"
+
+	"github.com/canonical/lxd/lxd/state"
+	"github.com/canonical/lxd/shared/api"
+)
+
+type OVNLogMonitor struct {
+	stateFunc func() *state.State
+	conn      net.PacketConn
+}
+
+// NewOVNLogMonitor returns a new OVNLogMonitor.
+func NewOVNLogMonitor(fn func() *state.State) *OVNLogMonitor {
+	if fn == nil {
+		panic("Invalid state function")
+	}
+
+	return &OVNLogMonitor{
+		stateFunc: fn,
+	}
+}
+
+// Start starts the log monitor. It listens for log messages using UDP, and sends them to the events API.
+func (l *OVNLogMonitor) Start(ctx context.Context, address string) (context.CancelFunc, error) {
+	var listenConfig net.ListenConfig
+	var err error
+
+	messages := make(chan string)
+
+	ctx, cancelFunc := context.WithCancel(ctx)
+
+	l.conn, err = listenConfig.ListenPacket(ctx, "udp", address)
+	if err != nil {
+		cancelFunc()
+		return nil, err
+	}
+
+	// This is the goroutine that processes the log messages. It also closes the connection if the context is done (this will terminate the second goroutine below as `ReadFrom` will receive an error).
+	go func() {
+		var applicationName string
```

Comment:
Why do you predefine these here rather than have per-iteration variables like you do with the `event` var itself?